packages/components/src/local/molecules/json-editor/index.tsx

```diff
@@ -190,18 +209,31 @@ export const JsonEditor: React.FC<Props> = ({
 
   useLayoutEffect(() => {
     if (editor) {
-      if (disabled) {
+      if (formId && !editButton) {
+        editor.disable()
+      } else if (disabled && !formId) {
         editor.disable()
       } else {
         editor.enable()
       }
     }
-  }, [editor])
+  }, [editor, editButton])
```

Comment:
@lilitkarapetyan - this method refers to `formId`.  Should it include `formId` in the hook props?

Reply:
Yes, because only components containing SaveMessageButton have a formId